entitlement/src/main/java/org/killbill/billing/entitlement/logging/EntitlementLoggingHelper.java

```diff
@@ -158,7 +159,7 @@ public static void logTransferEntitlement(final Logger log,
         }
     }
 
-    public static void logCancelEntitlement(final Logger log, final Entitlement entitlement, final LocalDate entitlementEffectiveDate, final Boolean overrideBillingEffectiveDate, final EntitlementActionPolicy entitlementPolicy, final BillingActionPolicy billingPolicy) {
+    public static void logCancelEntitlement(final Logger log, final Entitlement entitlement, final LocalDate entitlementEffectiveDate, final Boolean overrideBillingEffectiveDate, final EntitlementActionPolicy entitlementPolicy, final BillingActionPolicy billingPolicy, final DateTime entitlementEffectiveDateTime, final DateTime billingEffectiveDateTime) {
```

Comment:
Instead of introducing new arguments, maybe use `Serializable` for the dates:

```
[...], final Object entitlementEffectiveDate, final Object billingEffectiveDateTime, final Boolean overrideBillingEffectiveDate, [...]
```

I couldn't find a common ancestor between `LocalDate` and `DateTime` except `Serializable` unfortunately.

Reply:
Fixed